### PR TITLE
Implement resetting all permissions together with resetting accounts 

### PR DESCRIPTION
**Pull request by keshan3262**

### CODE REVIEW

File: src/store/root-state.epics.ts [hunk 1]
```diff
@@ -22,7 +24,11 @@ const rootStateResetEpic = (action$: Observable<Action>, state$: Observable<Root
     switchMap(keychainOptionsArray =>
       from(keychainOptionsArray).pipe(switchMap(options => Keychain.resetGenericPassword(options)))
     ),
-    mapTo(rootStateResetAction.success())
+    switchMap(() => from(BeaconHandler.getPermissions())),
+    switchMap(permissions =>
+      from(Promise.all(permissions.map(permission => BeaconHandler.removePermission(permission.accountIdentifier))))
+    ),
+    switchMap(() => [rootStateResetAction.success(), resetAllPermissionsAction()])
```
Comment (unixvb) [inline src/store/root-state.epics.ts:31]: you don't need `resetAllPermissionsAction` at all. `rootStateResetAction.success` will reset everything to the initial state (take look at `rootStateReducer`)

File: src/store/root-state.epics.ts [hunk 2]
```diff
@@ -22,7 +24,11 @@ const rootStateResetEpic = (action$: Observable<Action>, state$: Observable<Root
     switchMap(keychainOptionsArray =>
       from(keychainOptionsArray).pipe(switchMap(options => Keychain.resetGenericPassword(options)))
     ),
-    mapTo(rootStateResetAction.success())
+    switchMap(() => from(BeaconHandler.getPermissions())),
+    switchMap(permissions =>
+      from(Promise.all(permissions.map(permission => BeaconHandler.removePermission(permission.accountIdentifier))))
```
Comment (unixvb) [inline src/store/root-state.epics.ts:29]: use `forkJoin` instead of `Promise.all`

Comment (keshan3262) [reply src/store/root-state.epics.ts:29]: This process is about to be refactored, so there will be no need in it

File: src/store/root-state.epics.ts [hunk 3]
```diff
@@ -22,6 +23,7 @@ const rootStateResetEpic = (action$: Observable<Action>, state$: Observable<Root
     switchMap(keychainOptionsArray =>
       from(keychainOptionsArray).pipe(switchMap(options => Keychain.resetGenericPassword(options)))
     ),
+    switchMap(() => from(BeaconHandler.removeAllPermissions())),
```
Comment (unixvb) [inline src/store/root-state.epics.ts:26]: would this work in the same way without `from`?

**Review by andrewkryshtal — APPROVED** [review]

**Review by andrewkryshtal — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

**Pull request merged**